backend/dataall/modules/dataset_sharing/db/share_object_repositories.py

```diff
@@ -774,14 +774,20 @@ def find_all_share_items(session, share_uri, share_type):
         )
 
     @staticmethod
-    def other_approved_share_object_exists(session, environment_uri, dataset_uri):
+    def other_approved_share_item_table_exists(session, environment_uri, item_uri, share_item_uri):
+        share_item_shared_states = ShareItemSM.get_share_item_shared_states()
         return (
             session.query(ShareObject)
+            .join(
+                ShareObjectItem,
+                ShareObject.shareUri == ShareObjectItem.shareUri,
+            )
             .filter(
                 and_(
-                    Environment.environmentUri == environment_uri,
-                    ShareObject.status == ShareObjectStatus.Approved.value,
-                    ShareObject.datasetUri == dataset_uri,
+                    ShareObject.environmentUri == environment_uri,
+                    ShareObjectItem.itemUri == item_uri,
+                    ShareObjectItem.shareItemUri != share_item_uri,
+                    ShareObjectItem.status.in_(share_item_shared_states),
                 )
             )
             .all()
```

Comment:
A very minor improvement - Instead of returning all, we could use .first() and return the first appearance only for the case we have a lot of shares and share object items

Reply:
good call - resolved